docs/apache-airflow/authoring-and-scheduling/datasets.rst

```diff
@@ -230,8 +230,8 @@ Example:
         @task
         def print_triggering_dataset_events(triggering_dataset_events=None):
             for dataset, dataset_list in triggering_dataset_events.items():
-                print(dataset, dataset_list, dataset_list[dataset])
-                print(dataset_list[dataset][0].source_dag_run.dag_run_id)
+                print(dataset, dataset_list)
+                print(dataset_list[0].source_dag_run.dag_id)
```

Comment:
Both are correct:

Running:
```
        for dataset, dataset_list in triggering_dataset_events.items():
            print(dataset, dataset_list)
            print(f"{dataset_list[0].source_dag_run.dag_id=}")
            print(f"{dataset_list[0].source_dag_id=}")
```

results in:

```
[2023-10-18, 19:34:38 UTC] {logging_mixin.py:151} INFO - local://some_file.txt [DatasetEvent(id=6, dataset_id=6, extra={}, source_task_id='producer_task', source_dag_id='producer_dag', source_run_id='manual__2023-10-18T19:34:37.507108+00:00', source_map_index=-1)]
[2023-10-18, 19:34:38 UTC] {logging_mixin.py:151} INFO - dataset_list[0].source_dag_run.dag_id='producer_dag'
[2023-10-18, 19:34:38 UTC] {logging_mixin.py:151} INFO - dataset_list[0].source_dag_id='producer_dag'
```

(Granted my test dag is `producer_dag` and not `load_snowflake_data` from the demo

Reply:
Aaah, yes, you are right. I over-looked /home/jscheffl/Workspace/airflow/airflow/models/dataset.py:304 where the relation is modelled. Danger-zone is that from typing it can be also class of type `DatasetEventPydantic` if airflow switches to use internal API and this does not carry the relation parameter. Internal API is coming, adding a bit of more API complexity here for the future benefit of multi tenancy.